### PR TITLE
Implement support for Flutter screenshot

### PR DESCRIPTION
**Pull request by timsneath**
Implementing https://github.com/Dart-Code/Dart-Code/issues/657. 

OK - this is my first attempt at this. Be gentle :) but also feel free to send me back to the drawing board if I'm missing the mark! Remember -- I'm just a manager :)

### CODE REVIEW

**Comment by timsneath** [comment]
Fix for https://github.com/Dart-Code/Dart-Code/issues/657

File: src/commands/sdk.ts [hunk 1]
```diff
@@ -42,6 +42,13 @@ export class SdkCommands {
 			else
 				return this.runPub("get", uri);
 		}));
+		context.subscriptions.push(vs.commands.registerCommand("flutter.screenshot", async (uri) => {
+			if (!uri || !(uri instanceof Uri))
+				uri = await this.getWorkspace("Select which folder to save screenshot into");
```
Comment (DanTup) [inline src/commands/sdk.ts]: I think we should use a normal folder picker here (similar to [here](https://github.com/Dart-Code/Dart-Code/blob/76d5f6827ee432e29902f4d60560768402576dcb/src/sdk/utils.ts#L95)) so the user can put screenshots anywhere, not just in workspace roots (eg. they might want to put them on the desktop or on a screenshots sub-folder).

**Comment by DanTup** [comment]
Haha, it looks fine to me - though I added one comment about the folder.

I think there are two extra things we could do to make this better if you're interested:

1. Stash the folder in a variable so we don't prompt repeatedly if they want to take many screenshots
2. Add a (resource-scoped) setting (see `config.ts`) so if a user wants, they can store a screenshot path and not be prompted at all

**Comment by DanTup** [comment]
Apparently this branch has conflicts now; I'm happy to fix up and merge if you want (and also make the changes mentioned above), but I'll wait to hear from you first in case you want to do them!

**Comment by timsneath** [comment]
Oh, please don't hold features back for me. I'm tinkering really -- I figured this was a low-pri feature I could dabble with. Feel free to fix up and merge if you want to get this feature checked in and I'm not moving fast enough for you.

**Comment by DanTup** [comment]
No rush (users can run `flutter screenshot` in the terminal), just wanted to make sure you weren't too busy to come back to it. We can ship it whenever it's done 👍

**Comment by Phrohdoh** [comment]
> Add a (resource-scoped) setting (see config.ts) so if a user wants, they can store a screenshot path and not be prompted at all

👍 that would be fantastic! I often have a `screenshots` directory in the root of my projects and being able to write directly to that once configured would save me a few seconds here and there. 😄 

Thanks for your work on this @timsneath, glad to see progress being made (but, as Dan said, no rush of course)!

**Pull request closed without merging**